widgets/index-selector/index.js

```diff
@@ -0,0 +1,43 @@
+var React = require('react');
+var cx = require('classnames');
+
+var findIndex = require('lodash/array/findIndex');
+var utils = require('../../lib/widgetUtils.js');
+
+function indexSelector({
+    container = null,
+    cssClass = null,
+    indices = null
+  }) {
+  var IndexSelector = require('../../components/IndexSelector');
+  var containerNode = utils.getContainerNode(container);
+
+  var usage = 'Usage: indexSelector({container, indices[, cssClass]})';
+  if (container === null || indices === null) {
+    throw new Error(usage);
+  }
+
+  return {
+    init: function(state, helper) {
+      var currentIndex = helper.getIndex();
+      var isIndexInList = findIndex(indices, {'name': currentIndex}) !== -1;
+      if (!isIndexInList) {
+        throw new Error('[stats]: Index ' + currentIndex + ' not present in `indices`');
+      }
+    },
+
+    render: function(results, state, helper) {
+      React.render(
+        <IndexSelector
+          cssClass={cx(cssClass)}
+          currentIndex={helper.getIndex()}
+          indices={indices}
+          setIndex={helper.setIndex.bind(helper)}
```

Comment:
I have started doing in #63:

```
setIndex={setIndex}

// .. later on

function setIndex(helper, indexName) {
  helper.setIndex().search();
}
```

So that the component only knows about setIndex and do not have do do the search itself.

What do you think?

https://github.com/algolia/instantsearch.js/blob/feat/menu-1st-version/widgets/menu.js#L57-L70